src/utils.js

```diff
@@ -3,9 +3,12 @@ export function urlFormat(url, {filename=null, version=null}={}) {
     throw new Error('ArgumentError: File and version are required.');
   }
 
-  // This allows us to process jQuery's URLs.
-  return url.replace('$VERSION', version)
-            .replace('$FILENAME', filename)
-            .replace('$VERSION', version)
-            .replace('$FILENAME', filename);
+  var finalURL = url;
+  // Both the url and $FILENAME could $VERSION several times
+  while (finalURL.includes('$VERSION') || finalURL.includes('$FILENAME')) {
```

Comment:
On first sight, this seems to potentially become an infinite loop, however `$VERSION` won't contain `$FILENAME`, so it will break eventually.